### PR TITLE
Sudo ldap  netgroup_query bug fix patch

### PR DESCRIPTION
**Pull request by AZaugg**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [ ] The toolchain has been rebuilt successfully (or no changes were made to it)
- [ ] The toolchain/worker package manifests are up-to-date
- [ ] Any updated packages successfully build (or no packages were changed)
- [ ] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [ ] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [ ] All package sources are available
- [ ] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [ ] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [ ] All source files have up-to-date hashes in the `*.signatures.json` files
- [ ] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [ ] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

sudo backed by ldap is no longer working after moving to the new version of sudo. Patched the fix upstream to get ldap sudo to work with NETGROUP_QUERY 

###### Change Log  

- Adding upstream patch to Azure Linux version of sudo

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-YYYY-XXXX

###### Test Methodology

- I've not tested this yet

### CODE REVIEW

File: SPECS/sudo/sudo-ldap-netgroup-query.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/plugins/sudoers/ldap_conf.c b/plugins/sudoers/ldap_conf.c
```
Comment (christopherco) [inline SPECS/sudo/sudo-ldap-netgroup-query.patch:1]: Let's use the commit that supersedes the original change - https://github.com/sudo-project/sudo/commit/5fbf7a3625cc199e8f350af88c1120e45fa17bf4
Will help with this change

File: SPECS/sudo/sudo.signatures.json [hunk 2]
```diff
@@ -1,5 +1,6 @@
 {
    "Signatures": {
-      "sudo-1.9.14p3.tar.gz": "a08318b1c4bc8582c004d4cd9ae2903abc549e7e46ba815e41fe81d1c0782b62"
+      "sudo-1.9.14p3.tar.gz": "a08318b1c4bc8582c004d4cd9ae2903abc549e7e46ba815e41fe81d1c0782b62",
+      "sudo-ldap-netgroup-query.patch": "a830889178e984859b13a07b5ef458107794ca8683248bd4051403eae714dad6"
```
Comment (christopherco) [inline SPECS/sudo/sudo.signatures.json:4]: nit - patch file hash not needed

```suggestion
      "sudo-1.9.14p3.tar.gz": "a08318b1c4bc8582c004d4cd9ae2903abc549e7e46ba815e41fe81d1c0782b62"
```

File: SPECS/sudo/sudo.spec [hunk 3]
```diff
@@ -99,6 +100,9 @@ fi
 %exclude  /etc/sudoers.dist
 
 %changelog
+* Tue Dec 19  2023 Andy Zaugg <azaugg@linkedin.com> - 1.9.14p3-2
```
Comment (christopherco) [inline SPECS/sudo/sudo.spec:103]: nit - spacing

```suggestion
* Tue Dec 19 2023 Andy Zaugg <azaugg@linkedin.com> - 1.9.14p3-2
```

File: SPECS/sudo/sudo.spec [hunk 4]
```diff
@@ -1,13 +1,14 @@
 Summary:        Sudo
 Name:           sudo
 Version:        1.9.14p3
-Release:        1%{?dist}
+Release:        2%{?dist}
```
Comment (mfrw) [inline SPECS/sudo/sudo.spec:4]: Test Build: [PR-7068](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=471807&view=results)

**Comment by christopherco** [comment]
Requeued test build - https://dev.azure.com/mariner-org/mariner/_build/results?buildId=471817&view=results

**Review by christopherco — APPROVED** [review]

**Pull request merged**